File: task2.js
let checkPalindrome =
    (inputString, trashSymbolString) => {
        
        for (
            let i = 0, j = inputString.length - 1;
            i < Math.floor((inputString.length) / 2);
            i++, j--
        ) {
            while (trashSymbolString.includes(inputString[i])) {
                i++;
            }
            while (trashSymbolString.includes(inputString[j])) {
                j--;
            }
            if (inputString[i] !== inputString[j]) return false
        }

        return true;
    }


console.log(checkPalindrome("a@b!!b$a", "!@$"));
console.log(checkPalindrome("?Aaba#c", "#?"));

File: Task2.cs
using System;

namespace task2
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(checkPalindrome("a@b!!b$a", "!@$"));
            Console.WriteLine(checkPalindrome("?Aa#c", "#?"));
        }
        public static bool checkPalindrome(string inputString, string trashSymbolString)
        {
            int length = inputString.Length;
            for (int i = 0, j = length - 1; i < length; i++, j--)
            {
                while (trashSymbolString.Contains(inputString[i])) i++;
                while (trashSymbolString.Contains(inputString[j])) j--;
                if (inputString[i].CompareTo(inputString[j])!=0) return false;
            }

            return true;
        }

    }
}
